synfig-studio/src/gui/trees/layertree.cpp

```diff
@@ -225,6 +238,169 @@ LayerTree::create_layer_tree()
 	layer_tree_view().show();
 }
 
+auto t_start=std::chrono::high_resolution_clock::now();
+auto t_end=std::chrono::high_resolution_clock::now();
+
+bool
+LayerTree::on_param_tree_view_key_press_event(GdkEventKey* event)
+{
+	if(param_mouse_edit && (event->keyval == GDK_KEY_Escape))
+	{
+		//deprecated way of getting root window (in case we are not in parameter window)to set a transitional default cursor
+		//in the period between escape press and button release (after that the windows cursor is reset). is it ok to remove?
+		Glib::RefPtr<Gdk::Display> disp = Gdk::Display::get_default();
+		Glib::RefPtr<Gdk::Screen> scrn = disp->get_default_screen();
+		Glib::RefPtr<Gdk::Window> window = scrn->get_root_window();
+		window->set_cursor(window->get_cursor());
+
+		param_tree_view().get_window()->set_cursor(default_cursor);
+		synfigapp::Action::System::block_new_history=true;
+		just_finished=false;
+		value_base.set<float>(initial_value); row[param_model.value] = value_base;
+		synfigapp::Action::System::block_new_history=false;
+		while(gtk_events_pending ()) { gtk_main_iteration (); }
+	}
+	return false;
+}
+bool
+LayerTree::on_motion_notify_event(GdkEventMotion* event)
+{
+
+	auto t_now = std::chrono::high_resolution_clock::now();
+	double elapsed_time_ms = std::chrono::duration<double, std::milli>(t_now-t_start).count();
```

Comment:
Why are you trying to implement the click behavior yourself? Why not just use standard event listeners?

Reply:
I couldn't find anything in the standard listeners regarding the normal clicked event, closest I found was a long click gesture which I believe was also more geared toward touch devices. If there is one though, yes it would be much better.

Reply:
`signal_button_press_event`?

Reply:
Ok, I need a time to make a test app to test this.

Reply:
Alright, sounds good.